src/Handlers/RedisHandler.php

```diff
@@ -96,23 +105,23 @@ public function push(string $queue, string $job, array $data): bool
      */
     public function pop(string $queue, array $priorities): ?QueueJob
     {
-        $tasks = [];
-        $now   = Time::now()->timestamp;
-
-        foreach ($priorities as $priority) {
-            if ($tasks = $this->redis->zRangeByScore("queues:{$queue}:{$priority}", '-inf', (string) $now, ['limit' => [0, 1]])) {
-                if ($this->redis->zRem("queues:{$queue}:{$priority}", ...$tasks)) {
-                    break;
-                }
-                $tasks = [];
-            }
-        }
+        $now = Time::now()->timestamp;
+
+        // Prepare the arguments for the Lua script
+        $args = [
+            'queues:' . $queue,       // KEYS[1]
+            $now,                     // ARGV[2]
+            json_encode($priorities), // ARGV[3]
+        ];
+
+        // Execute the Lua script
+        $task = $this->redis->eval($this->luaScript, $args, 1);
 
-        if ($tasks === []) {
+        if ($task === false) {
```

Comment:
just to confirm, redis->eval can return false, while Predis ->eval can return null ?

Reply:
Funny enough - yes. This is all covered by tests.